truss-chains/truss_chains/stub.py

```diff
@@ -1,33 +1,222 @@
 import abc
 import asyncio
+import builtins
 import contextlib
 import contextvars
+import json
 import logging
-import ssl
+import sys
+import textwrap
 import threading
 import time
+import traceback
 from typing import (
     Any,
     AsyncIterator,
     ClassVar,
+    Dict,
     Iterator,
     Mapping,
+    NoReturn,
     Optional,
     Type,
     TypeVar,
+    Union,
     final,
+    overload,
 )
 
 import aiohttp
+import fastapi
 import httpx
+import pydantic
 import starlette.requests
 import tenacity
+from truss.templates.shared import serialization
 
 from truss_chains import definitions, utils
 
 DEFAULT_MAX_CONNECTIONS = 1000
 DEFAULT_MAX_KEEPALIVE_CONNECTIONS = 400
 
+
+_RetryPolicyT = TypeVar("_RetryPolicyT", tenacity.AsyncRetrying, tenacity.Retrying)
+_InputT = TypeVar("_InputT", pydantic.BaseModel, Any)  # Any signifies "JSON".
+_OutputT = TypeVar("_OutputT", bound=pydantic.BaseModel)
+
+
+# Error Propagation Utils. #############################################################
```

Comment:
Maybe better to move these utils to a separate file?

Reply:
I'm gonna think about moving some things around in general, like separating code that is intended to only run in the remote deployments better from the local/CLI/client code.

Let's for now leave it here.